### PR TITLE
Several minor improvements of evernote plugin

### PR DESCRIPTION
**Pull request by Hzj-jie**
Besides of fixing of #2539, this change includes,
1. A confirm window after purging history records to inquire users whether KOReaderClippings.txt should be removed also.
2. Exporting book notes in a single document to KOReaderClippings.txt is not reasonable: it does not use clippings history, and will mess up the KOReaderClippings.txt file.
3. Save settings before exporting all nodes in doc mode, to ensure book notes of current document have been saved.
4. "Notes can be found in ..." is not reasonable if nothing was exported.
5. Update text output format to avoid to use special characters of crengine. The KOReaderClippings.txt is rendered better in KOReader now.
6. Use KOReaderClippings.txt modification time to avoid duplications.

### CODE REVIEW

File: plugins/evernote.koplugin/main.lua [hunk 1]
```diff
@@ -162,8 +164,13 @@ function EvernoteExporter:addToMainMenu(tab_item_table)
                 text = _("Purge history records"),
                 callback = function()
                     self.config:purge()
-                    UIManager:show(InfoMessage:new{
-                        text = _("History records have been purged.\nAll notes will be exported again next time.\nIt is recommended to remove the existing KOReaderClipping.txt file to avoid duplication."),
+                    UIManager:show(ConfirmBox:new{
+                        text = _("History records have been purged.\nAll notes will be exported again next time.\nWould you like to remove existing KOReaderClipping.txt file to avoid duplication?\nRecords will be appended to KOReaderClipping.txt instead of being overwritten."),
```
Comment (Frenzie) [inline plugins/evernote.koplugin/main.lua]: "Would you like to remove existing KOReaderClipping.txt" ← missing "the" ;-)

Comment (Hzj-jie) [inline plugins/evernote.koplugin/main.lua]: :+1:

File: plugins/evernote.koplugin/main.lua [hunk 2]
```diff
@@ -463,23 +474,29 @@ function EvernoteExporter:exportBooknotesToHTML(title, booknotes)
...
-                file:write(T(_("   -- Page: %1, added on %2\n\n"),
-                             clipping.page, os.date("%c", clipping.time)))
-                if clipping.text then
-                    file:write(clipping.text)
-                end
-                if clipping.image then
-                    file:write(_("<An image>"))
+                -- If this clipping has already been exported, we ignore it.
+                if clipping.time >= file_modification then
+                    file:write(T(_(wide_space .. wide_space .. "-- Page: %1, added on %2\n"),
```
Comment (houqp) [inline plugins/evernote.koplugin/main.lua]: you will have to move the wide_space concatenation outside of `_()`, otherwise the translation won't work.

**Comment by houqp** [comment]
How does crengine render regular space in this case? I wonder if we should change crengine or here.

**Comment by Hzj-jie** [comment]
It simply ignore a regular space, and for "-", it displays "***" instead. It also ignores new line character after "_" or "+" or "*", I remember. This happens only on text file, AFAICT.

File: plugins/evernote.koplugin/main.lua (hunk 2 above)
Comment (Hzj-jie) [inline plugins/evernote.koplugin/main.lua]: Updated.

**Comment by houqp** [comment]
CREngine's txt rendering heuristics is indeed very strange. I wish it just render the file as is without trying to be "smart".

**Comment by houqp** [comment]
Looks like there is a conflict to the current master, can you help resolve it?

**Comment by Hzj-jie** [comment]
The conflict has been resolved.

**Comment by houqp** [comment]
Weird, github still prevents me from merging this PR due to conflicts

**Comment by Hzj-jie** [comment]
Works :)

**Pull request merged**